src/posix/platform/infra_if.cpp

```diff
@@ -182,73 +178,41 @@ void platformInfraIfInit(otInstance *aInstance, const char *aIfName)
 
     // Initializes the infra interface.
     ifIndex = if_nametoindex(aIfName);
-    if (ifIndex == 0)
-    {
-        otLogCritPlat("failed to get the index for infra interface %s: %s", aIfName, strerror(errno));
-        DieNow(OT_EXIT_ERROR_ERRNO);
-    }
+    VerifyOrDie(ifIndex != 0, OT_EXIT_ERROR_ERRNO);
```

Comment:
I think it is valuable to have more friendly error output for common user-facing issues (e.g. when specifying an interface that cannot be found). As an example, we added some more user-friendly error output in #5935.

Reply:
Agree with @jwhui . 

I think it's better to allow specifying an error message in `VerifyOrDie`. 

e.x.  
```cpp
VerifyOrDie(ifIndex != 0, OT_EXIT_ERROR_ERRNO, "failed to get the index for infra interface %s: %s", aIfName, strerror(errno));
```

Reply:
Makes sense. reverted the change to the interface name check.